yt_dlp/extractor/nubilesporn.py

```diff
@@ -0,0 +1,143 @@
+import re
+from typing import Optional
+
+from .common import InfoExtractor
+from ..utils import parse_resolution, \
+    get_element_text_and_html_by_tag, get_element_html_by_class, get_elements_html_by_class, extract_attributes, \
+    clean_html, urlencode_postdata, get_element_by_class, float_or_none, int_or_none, \
+    get_element_by_id, get_elements_by_class, unified_timestamp
+
+
+class NubilesPornIE(InfoExtractor):
+    _NETRC_MACHINE: str = 'nubiles-porn'
+
+    _BASE_DOMAIN: str = 'nubiles-porn.com'
+    _BASE_URL: str = f'https://{_BASE_DOMAIN}'
+    _MEMBERS_URL = f'https://members.{_BASE_DOMAIN}'
+
+    _VALID_URL: str = r'https://members.nubiles-porn.com/video/watch/(?P<id>\d+)/(?P<display_id>[a-z\d\-]+-(s(?P<season>\d+)e(?P<episode>\d+))$)'
+    _VALID_VIDEO_URL: str = f're:https://content2a.{_BASE_DOMAIN}/exclusive/' + r'(?:[a-z_]+)/videos/(?:[a-z\d_]+).mp4\?st=(?:[\w\-]{22})&e=(?:[0-9]{10})&tr=(?:[A-Z0-9]{154})'
+
+    _TESTS = [{
+        'url': f'{_MEMBERS_URL}/video/watch/165320/trying-to-focus-my-one-track-mind-s3e1',
+        'md5': 'fa7f09da8027c35e4bdf0f94f55eac82',
+        'info_dict': {
+            'id': '165320',
+            'title': 'Trying To Focus My One Track Mind - S3:E1',
+            'formats': [
+                {'url': _VALID_VIDEO_URL, 'format_id': 'mp4_480_270', 'width': 480, 'height': 270},
+                {'url': _VALID_VIDEO_URL, 'format_id': 'mp4_640_360', 'width': 640, 'height': 360},
+                {'url': _VALID_VIDEO_URL, 'format_id': 'mp4_960_540', 'width': 960, 'height': 540},
+                {'url': _VALID_VIDEO_URL, 'format_id': 'mp4_1280_720', 'width': 1280, 'height': 720},
+                {'url': _VALID_VIDEO_URL, 'format_id': 'mp4_1920_1080', 'width': 1920, 'height': 1080},
+                {'url': _VALID_VIDEO_URL, 'format_id': 'mp4_3840_2160', 'width': 3840, 'height': 2160}
+            ],
```

Comment:
Tests aren't meant to check the format URLs. You can keep the check for format metadata if u think it'll remain consistent, but it isn't needed imo

Tip: Remove all fields except ext/id and run the test. It'll tell you which fields are needed.